wiring/src/spark_wiring_ble.cpp

```diff
@@ -700,34 +748,58 @@ String BleAdvertisingData::deviceName() const {
 
 size_t BleAdvertisingData::serviceUUID(BleUuid* uuids, size_t count) const {
     size_t found = 0;
-    found += serviceUUID(BleAdvertisingDataType::SERVICE_UUID_16BIT_MORE_AVAILABLE, &uuids[found], count - found);
-    found += serviceUUID(BleAdvertisingDataType::SERVICE_UUID_16BIT_COMPLETE, &uuids[found], count - found);
-    found += serviceUUID(BleAdvertisingDataType::SERVICE_UUID_128BIT_MORE_AVAILABLE, &uuids[found], count - found);
-    found += serviceUUID(BleAdvertisingDataType::SERVICE_UUID_128BIT_COMPLETE, &uuids[found], count - found);
+    const Vector<BleUuid>& foundUuids = serviceUUID();
+    for (const auto& uuid : foundUuids) {
+        if (found >= count) {
+            break;
+        }
+        uuids[found++] = uuid;
+    }
     return found;
 }
 
+Vector<BleUuid> BleAdvertisingData::serviceUUID() const {
+    Vector<BleUuid> foundUuids;
+    serviceUUID(BleAdvertisingDataType::SERVICE_UUID_16BIT_MORE_AVAILABLE, foundUuids);
+    serviceUUID(BleAdvertisingDataType::SERVICE_UUID_16BIT_MORE_AVAILABLE, foundUuids);
+    serviceUUID(BleAdvertisingDataType::SERVICE_UUID_16BIT_COMPLETE, foundUuids);
+    serviceUUID(BleAdvertisingDataType::SERVICE_UUID_128BIT_MORE_AVAILABLE, foundUuids);
+    serviceUUID(BleAdvertisingDataType::SERVICE_UUID_128BIT_COMPLETE, foundUuids);
+    return foundUuids;
+ }
+
 size_t BleAdvertisingData::customData(uint8_t* buf, size_t len) const {
     return get(BleAdvertisingDataType::MANUFACTURER_SPECIFIC_DATA, buf, len);
 }
 
+uint8_t BleAdvertisingData::operator[](uint8_t i) const {
+    if (i >= selfLen_) {
+        return 0;
+    }
+    return selfData_[i];
+}
+
 bool BleAdvertisingData::contains(BleAdvertisingDataType type) const {
     size_t adsOffset;
     return locate(selfData_, selfLen_, type, &adsOffset) > 0;
 }
 
-size_t BleAdvertisingData::serviceUUID(BleAdvertisingDataType type, BleUuid* uuids, size_t count) const {
+size_t BleAdvertisingData::serviceUUID(BleAdvertisingDataType type, Vector<BleUuid>& uuids) const {
```

Comment:
Nit: We can probably just return `Vector<BleUuid>` here. No need to separately track `found`. The vector has the size.